### DISCUSSION TITLE
SwitchNetwork  seems not working

### DISCUSSION THREAD

**Discussion by abhishekDabba** [reown-com/appkit#3240] [Q&A]
I have a custom test network (EVM Compatible ) for which I am using metamask btw the integration is for multichain 

while creating the instance of APPKIT I make sure to only provide the PeaqTestnet for compatibility but when I test by  connecting any Ethereumchain  the wallet is successfully getting connected  I want that it should request to switch network and make sure only user is able to connect through peaq chain if using metamask wallet

```import React, { useState, useEffect } from "react";
import { useAppKit, useAppKitEvents, useAppKitAccount, useDisconnect, useAppKitNetwork } from '@reown/appkit/react';
import { peaqTestnet } from '../../constants/network';

const CustomConnectButton = () => {
    const { open } = useAppKit();
    const { switchNetwork } = useAppKitNetwork();
    const { address, isConnected } = useAppKitAccount();
    const { disconnect } = useDisconnect();
    const { data: eventData } = useAppKitEvents();
    const [showDropdown, setShowDropdown] = useState(false);

    const toggleDropdown = () => setShowDropdown(!showDropdown);

    const enforceNetwork = async () => {
        if (isConnected) {
            console.log("ENFORCING NETWORK NOW");
            await switchNetwork(peaqTestnet);
            console.log("Network Switched to Peaq Testnet");
        }
    };

    useEffect(() => {
        if (eventData) {
            console.log("eventData.event:", eventData.event);

            switch (eventData.event) {
                case 'SELECT_WALLET':
                    console.log("Wallet selected:", eventData.properties?.name);
                    break;
                
                case 'CONNECT_SUCCESS':
                    console.log("Connected successfully!");
                    enforceNetwork(); // Enforce network upon successful connection
                    break;
                
                case 'DISCONNECT_SUCCESS':
                    console.log("Disconnected successfully!");
                    disconnect(); // Ensure disconnection
                    break;
                
                case 'SWITCH_NETWORK':
                    console.log("Switching network to enforce Peaq Testnet");
                    enforceNetwork(); // Re-enforce network if user tries to switch
                    break;

                case 'CONNECT_ERROR':
                    console.error("Connection error:", eventData.properties?.message);
                    break;
                
                default:
                    console.log("Unhandled event:", eventData);
                    break;
            }
        }
    }, [eventData, isConnected]);

    const buttonStyle = {
        backgroundColor: 'black',
        color: '#FFD903',
        border: '1px solid #FFD903',
        padding: '6px 12px',
        borderRadius: '8px',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
    };

    if (isConnected) {
        return (
            <div className="relative inline-block">
                <button onClick={toggleDropdown} style={buttonStyle}>
                    {`${address.slice(0, 6)}...${address.slice(-4)}`}
                </button>
                {showDropdown && (
                    <div className="absolute right-0 mt-2 w-40 bg-gray-800 text-white rounded-lg shadow-lg z-10">
                        <button
                            onClick={() => {
                                disconnect();
                                setShowDropdown(false);
                            }}
                            className="w-full px-4 py-2 text-left hover:bg-gray-700"
                            style={{ color: '#FFD903' }}
                        >
                            Disconnect
                        </button>
                    </div>
                )}
            </div>
        );
    }

    return (
        <button onClick={() => open()} style={buttonStyle}>
            Select Wallet
        </button>
    );
};

export default CustomConnectButton;
```

here's how I tried to use the SwitchNetwork but It seems not working the network is still Ethereum even after the function is running but not providing any such output that we expect !

**Comment by rtomas** [comment]
hi. Which version are you using ?

**Reply by abhishekDabba** [reply]
here's the versions 

```@reown/appkit-adapter-ethers": "^1.3.2",
    "@reown/appkit-adapter-solana": "^1.3.2",
    "@solana/spl-token": "^0.4.8",
    "@solana/wallet-adapter-react": "^0.15.35",
    "@solana/wallet-adapter-wallets": "^0.19.32",
    "@solana/web3.js": "^1.95.4",
    "ethers": "^6.13.4",
    ```

**Comment by rtomas** [comment]
hi @abhishekDabba 

Could you please try in latest ? v1.4.1 ?

I have just tried and is working for me.

Otherwise, please share "peaqTestnet" declaration and AppKit Configuration.

Thanks

**Reply by abhishekDabba** [reply]
sure here's my 
Kit Configuration 
```import { createAppKit } from '@reown/appkit/react';
import React from 'react';
import { SolanaAdapter } from '@reown/appkit-adapter-solana';
import { EthersAdapter } from '@reown/appkit-adapter-ethers';
import { peaqTestnet } from '.././constants/network';
import { solana, solanaDevnet } from '@reown/appkit/networks';
import { SolflareWalletAdapter, PhantomWalletAdapter } from '@solana/wallet-adapter-wallets';

// 0. Create the Ethers adapter
export const ethersAdapter = new EthersAdapter();

// 1. Create Solana adapter
const solanaWeb3JsAdapter = new SolanaAdapter({
  wallets: [new PhantomWalletAdapter(), new SolflareWalletAdapter()],
});

const projectId = '1a1-------------------2';

// Metadata for the AppKit modal
const metadata = {
  name: 'AppKit Example',
  description: 'AppKit Example for Wallet Connection',
  url: 'https://yourdomain.com',
  icons: ['https://avatars.githubusercontent.com/u/179229932'],
};

export const appKitInstance = createAppKit({
  adapters: [ethersAdapter, solanaWeb3JsAdapter],
  networks: [solana, solanaDevnet, peaqTestnet],
  metadata,
  projectId,
  features: {
      email: false, // Disable email login
      socials: [],  // Disable all social logins
      emailShowWallets: true, // Show wallet options directly on the initial screen
  },
  allWallets: 'SHOW',
  themeMode: 'dark',  // or 'dark' depending on your app's default theme
  themeVariables: {
      '--w3m-accent': '#000000', // Yellow color for text and border
      '--w3m-color-mix': '#1C1C1C', // Black background color
      '--w3m-color-mix-strength': 100, // Full black background
      '--w3m-border-radius-master': '1px', // Rounded corners for button
      '--w3m-font-family': 'Mulish, sans-serif', // Font family for text
      '--w3m-font-size-master': '0.65rem', // Font size for button text
  },
});


function WalletContextProvider({ children }) {
  return <>{children}</>;
}

export default WalletContextProvider;
```

Testnet Configuration
```import { defineChain } from '@reown/appkit/networks';

export const peaqTestnet = defineChain({
    id: 9990,
    caipNetworkId: 'eip155:9990',
    chainNamespace: 'eip155',
    name: 'Peaq Testnet',
    nativeCurrency: {
        decimals: 18,
        name: 'Agung',
        symbol: 'AGNG',
    },
    rpcUrls: {
        default: {
            http: ['https://rpcpc1-qa.agung.peaq.network/'],
        },
    },
    blockExplorers: {
        default: { name: 'Peaq Explorer', url: 'https://agung-testnet.subscan.io/' },
    },
});
```

**Reply by abhishekDabba** [reply]
hey sorry for the inconvenience this integration is taking some more time can you help us resolve this issues early we have deadlines for launch happy to use reown provider !

here's the list of issues :
- Account change does not reflect on UI or variable's like address 
- I am still able to connect to different EVM chains even if only I have passed in PEAQ as network (KIT configuration shared above)

hoping for complete response and help

**Reply by rtomas** [reply]
Can you share a screen video of the account change issue ?

i'll be looking this as soon as possible.

**Reply by abhishekDabba** [reply]
hey I prepared the loom showing the issues there 
[https://www.loom.com/share/b46a9b27b09b4e66a390e5ff8857d953?sid=b93be6db-76e2-42d3-bd9a-bf49aacc0e41](url)

**Reply by abhishekDabba** [reply]
any chance I can remove the ethereum option from here !
likewise remove solana if someone clicks on metamask. just to avoid confusions since we are having chain specific wallets.Images are shared below

<img width="403" alt="image" src="https://github.com/user-attachments/assets/c63ec3e3-fc81-4805-ae03-f5a4677b4cda">
<img width="393" alt="image" src="https://github.com/user-attachments/assets/545abac5-e03e-45ed-abba-b076cf56a002">

**Reply by rtomas** [reply]
hi @abhishekDabba 

so you want your users to use phantom for solana and metamask for evm ? that is what you are requesting ?

**Reply by rtomas** [reply]
> I am still able to connect to different EVM chains even if only I have passed in PEAQ as network (KIT configuration shared above)

This should be solved on v1.5.0 .. can you confirm ?

**Reply by abhishekDabba** [reply]
> This should be solved on v1.5.0 .. can you confirm ?

so the Network connections seems working where if I switch or connect via other network then it open a tab specifying that the network is not supported , but the modal is not clickable completely  I have shared a loom video for the same , when clicking on peaq testnet it does not work when clicking on solana it directly disconnects

[https://www.loom.com/share/a8adf95799154adfa2f1c73544243ee0?sid=55aab640-1898-4fe3-9e2b-f9bc69c37d5a](url)




> hi @abhishekDabba
> 
> so you want your users to use phantom for solana and metamask for evm ? that is what you are requesting ?

Yes to avoid confusion for users If we can streamline this and if someone clicks phantom we show only solana and if someone clicks metamask we show them Peaq option its Ethereum there that feels like mainnet but our chains is different can we make it PEAQ ?

**Reply by abhishekDabba** [reply]
hi @rtomas waiting for you response ! also my team wants to know in case we can use REOWN for our product (can the customizations be done to the wallet ?)

**Reply by abhishekDabba** [reply]
@arein can you help ?

**Reply by arein** [reply]
@abhishekDabba after upgrading which issue remains?

**Reply by abhishekDabba** [reply]
> @abhishekDabba after upgrading which issue remains?

so the Network connections seems working where if I switch or connect via other network then it open a tab specifying that the network is not supported , but the modal is not clickable completely I have shared a loom video for the same , when clicking on peaq testnet it does not work when clicking on solana it directly disconnects

[https://www.loom.com/share/a8adf95799154adfa2f1c73544243ee0?sid=55aab640-1898-4fe3-9e2b-f9bc69c37d5a](https://github.com/orgs/reown-com/discussions/url)




> hi @abhishekDabba
> 
> so you want your users to use phantom for solana and metamask for evm ? that is what you are requesting ?

Yes to avoid confusion for users If we can streamline this and if someone clicks phantom we show only solana and if someone clicks metamask we show them Peaq option its Ethereum there that feels like mainnet but our chains is different can we make it PEAQ ?

**Reply by rtomas** [reply]
I understand your feature request and will be sending it to our team.

was your issue solved ?

**Reply by abhishekDabba** [reply]
hi @rtomas 
the fix that you made in v1.5.0 seems working good only issue left in functionality part is 

<img width="373" alt="Screenshot 2024-11-26 at 2 05 00 PM" src="https://github.com/user-attachments/assets/cd6660d5-54d8-4d17-b4ee-dbdfd7f0a91b">

when I click on peaq it seems non clickable rest solana , devnet seems working perfectly can you lemme know why the peaq seems not clickable ?

rest as you answered above that the feature request is been handled by your team , looking forward to hear it anytime soon !

Thanks

**Reply by abhishekDabba** [reply]
@rtomas  any updates on why the peaq seems not clickable ?

**Reply by abhishekDabba** [reply]
@rtomas @arein still waiting for the response 🙂

**Reply by rtomas** [reply]
Not yet. Will be contacting you if we have news about this issue. 

Thanks for the patience

**Comment by 1012492240** [comment]
![image](https://github.com/user-attachments/assets/10f23883-127f-4fa6-a8cf-8d4c511235e4)

重复了

**Reply by abhishekDabba** [reply]
how can I get this  specific for chain instead of multi-chain ?

**Reply by rtomas** [reply]
@1012492240 which version you are using ?

**Comment by arein** [comment]
@abhishekDabba keen to get you unblocked - wanna DM me on Telegram? `cyberdrk`

**Reply by abhishekDabba** [reply]
sure Dmed you pls check !
